### ISSUE TITLE
job isRunning returning false after calling now

### ISSUE THREAD

**Issue by aedeveloper** [agenda/agenda#1139]
I think isRunning method doesn't work properly with now method.

because some attrs is missing `lockedAt` and `lastFinishedAt` from now method.

```
      const job = await agenda.now("test");

      job.isRunning() // false
```

the issue because this line does not comply with now attrs

https://github.com/agenda/agenda/blob/abfd348a8f1a1edccf255bbeab9dc3ff6f4e87f8/lib/job/is-running.ts#L21

**Comment by simllll** [comment]
lastRunAt should be set when the job starts running.
see https://github.com/agenda/agenda/blob/master/lib/job/run.ts#L19

the problem her eis something else, agenda picks up the job again from database and creates a new "job" object.. so the job object that is returned from agenda.now is not ident with the one that is processed by the job processor

**Comment by aedeveloper** [comment]
So in this case `isRunning` needs to pick up the updated object of the job.
otherwise that method doesn't make sense anymore.

**Comment by simllll** [comment]
Not exactly a bug, but not very well documented maybe.
Imagine follwing scenario:
You have two servers.. one is a frontend webserver and one is a backend (worker) server.
The frontend server just queus jobs (e.g. with agenda.now()), and the backend server picks up the jobs and processes them.

**fronted:**
`const job = agenda.now('bla');` 
this job has now only the document id, and therefore is kinda useless.

**backend:**
* picks up the job
* starts processing (now the job has "isRunning"
* .... processes the job.

so the "isRunning" flag would be valid on the backend, but never on the frontend, it is used by the job processor.

To get the real current state on the frontend, you would need to query the database again, e.g. do the following:

```ts
const jobs = await agenda.jobs({_id: job.attrs._id})
console.log(jobs[0].isRunning()
```

but beware, now you still have only one "snapshot" of the current state, if you want to watch the status, you need to query the database again (each time you ask if the job is running)

**Comment by aedeveloper** [comment]
My point is `agneda.now('bla')` already has the document id as it passing **this** to isRunning method.
if it just makes a new **query** within the method to get the updated object each time we call `job.isRunning()` and that will make more sense.
without any further lines.

**Comment by simllll** [comment]
I've impleented in @hokify/agenda. Would you like to test it and let me know if it works for you?
https://github.com/hokify/agenda/commit/1aaaa61f0a009563a84cb81036427c187076f190

NPM https://www.npmjs.com/package/@hokify/agenda

**Comment by aedeveloper** [comment]
it doesn't seem to work

the result is always **false**

**Comment by simllll** [comment]
Thanks, I just released v4.1.2 with a better implemention of this.
(Beware that isRunning returns a promise.)

Can you retry? :)

**Comment by aedeveloper** [comment]
Now works fine.

BTW, you may consider getJob() beside  getJobs() that will be a lot easier.
